src/extension.ts

```diff
@@ -9,22 +9,29 @@ import { CompletionProvider } from './providers';
 export let errorDiagnosticCollection = vscode.languages.createDiagnosticCollection("terraform-error");
 export let outputChannel = vscode.window.createOutputChannel("Terraform");
 
+const documentSelector: vscode.DocumentSelector = [
+    { language: "terraform", scheme: "file" },
+    { language: "terraform", scheme: "untitled" }
+];
+
 export function activate(ctx: vscode.ExtensionContext) {
     ctx.subscriptions.push(errorDiagnosticCollection);
 
     let formattingProvider = new FormattingEditProvider;
 
     // we need to manually handle save events otherwise format on autosave does not work
     ctx.subscriptions.push(vscode.workspace.onDidSaveTextDocument((doc) => formattingProvider.onSave(doc)));
```

Comment:
do we need some magic in the `onDidSaveTextDocument()` handler aswell?

Reply:
@mauve Ah good point! I'll move this subscription into the same conditional statement as the call to `initializeIndex`.